pkg/controller/node/nodecontroller.go

```diff
@@ -312,6 +356,9 @@ func NewNodeController(
 						utilruntime.HandleError(fmt.Errorf("Error allocating CIDR: %v", err))
 					}
 				}
+				if nc.taintManager != nil {
```

Comment:
I don't like the pattern of moving nil checks into the functions (if not necessary). I'll keep it on the higher level, guarded by `nc.taintManager != nil` everywhere which is not a create-related code.